packages/rich-text/src/register-format-type.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import { isFunction, isString } from 'lodash';
+import { default as tinycolor, mostReadable } from 'tinycolor2';
+
+/**
+ * WordPress dependencies
+ */
+import { select, dispatch } from '@wordpress/data';
+import { Component, isValidElement } from '@wordpress/element';
+
+/**
+ * Array of icon colors containing a color to be used if the icon color
+ * was not explicitly set but the icon background color was.
+ *
+ * @type {Object}
+ */
+const ICON_COLORS = [ '#191e23', '#f8f9f9' ];
+
+/**
+ * Function that checks if the parameter is a valid icon.
+ *
+ * @param {*} icon  Parameter to be checked.
+ *
+ * @return {boolean} True if the parameter is a valid icon and false otherwise.
+ */
+
+export function isValidIcon( icon ) {
```

Comment:
Ideally this needs to be shared with the `blocks` package,

Reply:
Can we use the one from blocks both `isValidIcon` and `normalizeIconObject`? Inline block is one of the special formatter options, so this would make sense. Any blockers?

Reply:
I'm not sure I'm understanding this. Should I import them from the blocks package?